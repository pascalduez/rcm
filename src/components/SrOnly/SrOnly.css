:root {
  /**
   * Only display content to screen readers.
   * @see: http://a11yproject.com/posts/how-to-hide-content
   * @see: http://hugogiraudel.com/2016/10/13/css-hide-and-seek
   */
  --sr-only {
    position: absolute;
    width: 1px;
    height: 1px;
    padding: 0;
    overflow: hidden;
    clip: rect(0, 0, 0, 0);
    white-space: nowrap;
    border: 0;
  }

  /**
   * Use in conjunction with .sr-only to only display content when it's focused.
   * Useful for "Skip to main content" links
   * @see http://www.w3.org/TR/2013/NOTE-WCAG20-TECHS-20130905/G1
   */
  --sr-only-focusable {
    &:active,
    &:focus {
      position: static;
      width: auto;
      height: auto;
      overflow: visible;
      clip: auto;
      white-space: normal;
    }
  }
}

.all {
  @apply --sr-only;
}

/*
Base em at 16px base font size: 0.0625em

20em (320px) 19.938
30em (480px) 29.938
37.5em (600px) 37.438
48em (768px) 47.938
56.250em (900px) 56.188
68.750em (1100px) 68.688
81.250em (1300px) 81.188
*/

@media screen and (max-width: 19.938em) {
  .xsm {
    @apply --sr-only;
  }
}

@media screen and (min-width: 20em) and (max-width: 29.938em) {
  .sm {
    @apply --sr-only;
  }
}

@media screen and (min-width: 30em) and (max-width: 37.438em) {
  .md {
    @apply --sr-only;
  }
}

@media screen and (min-width: 37.5em) and (max-width: 47.938em) {
  .lg {
    @apply --sr-only;
  }
}

@media screen and (min-width: 48em) and (max-width: 56.188em) {
  .xlg {
    @apply --sr-only;
  }
}

@media screen and (min-width: 56.25em) and (max-width: 68.688em) {
  .wd {
    @apply --sr-only;
  }
}

@media screen and (min-width: 68.75em) and (max-width: 81.188em) {
  .xwd {
    @apply --sr-only;
  }
}

@media screen and (min-width: 81.25em) {
  .xxwd {
    @apply --sr-only;
  }
}
