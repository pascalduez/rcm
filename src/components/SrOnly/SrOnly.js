// @flow

import * as React from 'react';
import classNames from 'classnames';

import styles from './SrOnly.css';

export type Props = {
  xsm?: boolean,
  sm?: boolean,
  md?: boolean,
  lg?: boolean,
  xlg?: boolean,
  xxlg?: boolean,
  wd?: boolean,
  xwd?: boolean,
  tagName?: string,
  className?: string,
  children: React.Node,
};

function SrOnly(props: Props) {
  const {
    xsm,
    sm,
    md,
    lg,
    xlg,
    xxlg,
    wd,
    xwd,
    tagName = 'span',
    className,
    children,
    ...extraProps
  } = props;

  const hasBreapoint = Object.values(props).some(val => val === true);

  const rootClasses = classNames(
    {
      [styles.all]: !hasBreapoint,
      [styles.xsm]: xsm,
      [styles.sm]: sm,
      [styles.md]: md,
      [styles.lg]: lg,
      [styles.xlg]: xlg,
      [styles.xxlg]: xxlg,
      [styles.wd]: wd,
      [styles.xwd]: xwd,
    },
    className
  );

  return React.createElement(
    tagName,
    { className: rootClasses, ...extraProps },
    children
  );
}

export default SrOnly;
