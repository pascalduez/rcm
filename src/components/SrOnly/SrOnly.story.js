// @flow

import * as React from 'react';

import Component from './SrOnly';

const SrOnly = props => <Component tagName="div" {...props} />;

export default () => (
  <div theme="grey" style={{ lineHeight: 2 }}>
    <SrOnly>Hidden at all breakpoints</SrOnly>
    <SrOnly xsm>Hidden at xsm breakpoint</SrOnly>
    <SrOnly sm>Hidden at sm breakpoint</SrOnly>
    <SrOnly md>Hidden at md breakpoint</SrOnly>
    <SrOnly lg>Hidden at lg breakpoint</SrOnly>
    <SrOnly xlg>Hidden at xlg breakpoint</SrOnly>
    <SrOnly xxlg>Hidden at xxlg breakpoint</SrOnly>
    <SrOnly wd>Hidden at wd breakpoint</SrOnly>
    <SrOnly xwd>Hidden at xwd breakpoint</SrOnly>
  </div>
);
