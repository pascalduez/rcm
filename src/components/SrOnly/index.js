// @flow

export { default } from './SrOnly';
export type { Props as SrOnlyProps } from './SrOnly';
