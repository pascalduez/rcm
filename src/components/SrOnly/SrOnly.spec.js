// @flow

import React from 'react';
import { render } from 'enzyme';

import SrOnly from './SrOnly';

describe('<SrOnly />', () => {
  it('should render the proper default markup and classes', () => {
    const wrapper = render(<SrOnly>test</SrOnly>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render the proper `xsm` breakpoint classes', () => {
    const wrapper = render(<SrOnly xsm>test</SrOnly>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render the proper `sm` breakpoint classes', () => {
    const wrapper = render(<SrOnly sm>test</SrOnly>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render the proper `md` breakpoint classes', () => {
    const wrapper = render(<SrOnly md>test</SrOnly>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render the proper `lg` breakpoint classes', () => {
    const wrapper = render(<SrOnly lg>test</SrOnly>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render the proper `xlg` breakpoint classes', () => {
    const wrapper = render(<SrOnly xlg>test</SrOnly>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render the proper `xxlg` breakpoint classes', () => {
    const wrapper = render(<SrOnly xxlg>test</SrOnly>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render the proper `wd` breakpoint classes', () => {
    const wrapper = render(<SrOnly wd>test</SrOnly>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render the proper `xwd` breakpoint classes', () => {
    const wrapper = render(<SrOnly xwd>test</SrOnly>);

    expect(wrapper).toMatchSnapshot();
  });
});
