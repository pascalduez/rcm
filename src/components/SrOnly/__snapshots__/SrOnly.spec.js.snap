// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SrOnly /> should render the proper \`lg\` breakpoint classes 1`] = `
<span
  class="SrOnly-lg"
>
  test
</span>
`;

exports[`<SrOnly /> should render the proper \`md\` breakpoint classes 1`] = `
<span
  class="SrOnly-md"
>
  test
</span>
`;

exports[`<SrOnly /> should render the proper \`sm\` breakpoint classes 1`] = `
<span
  class="SrOnly-sm"
>
  test
</span>
`;

exports[`<SrOnly /> should render the proper \`wd\` breakpoint classes 1`] = `
<span
  class="SrOnly-wd"
>
  test
</span>
`;

exports[`<SrOnly /> should render the proper \`xlg\` breakpoint classes 1`] = `
<span
  class="SrOnly-xlg"
>
  test
</span>
`;

exports[`<SrOnly /> should render the proper \`xsm\` breakpoint classes 1`] = `
<span
  class="SrOnly-xsm"
>
  test
</span>
`;

exports[`<SrOnly /> should render the proper \`xwd\` breakpoint classes 1`] = `
<span
  class="SrOnly-xwd"
>
  test
</span>
`;

exports[`<SrOnly /> should render the proper \`xxlg\` breakpoint classes 1`] = `
<span
  class="undefined"
>
  test
</span>
`;

exports[`<SrOnly /> should render the proper default markup and classes 1`] = `
<span
  class="SrOnly-all"
>
  test
</span>
`;
